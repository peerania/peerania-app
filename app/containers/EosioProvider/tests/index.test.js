import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import configureStore from 'configureStore';
import createdHistory from 'createdHistory';

import { EosioProvider } from '../index';

const child = <div>children</div>;
React.Children.only = jest.fn().mockImplementation(() => child);

// const cmp = new EosioProvider();
const props = {
  initializing: false,
  initEosio: jest.fn(),
};

describe('<EosioProvider />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, createdHistory);
  });

  describe('should render and match the snapshot', () => {
    it('initializing is false', () => {
      props.initializing = false;
      const {
        container: { firstChild },
      } = render(
        <Provider store={store}>
          <EosioProvider {...props} />
        </Provider>,
      );
      expect(firstChild).toMatchSnapshot();
    });

    it('initializing is true', () => {
      props.initializing = true;
      const {
        container: { firstChild },
      } = render(
        <Provider store={store}>
          <EosioProvider {...props} />
        </Provider>,
      );
      expect(firstChild).toMatchSnapshot();
    });
  });

  /*it('componentDidMount', () => {
    cmp.componentDidMount();
    expect(cmp.props.initEosio).toHaveBeenCalled();
  });*/
});
