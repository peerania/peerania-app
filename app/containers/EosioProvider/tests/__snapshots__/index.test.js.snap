// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EosioProvider /> should render and match the snapshot initializing is false 1`] = `
<div>
  children
</div>
`;

exports[`<EosioProvider /> should render and match the snapshot initializing is true 1`] = `
<div>
  children
</div>
`;
